controller/src/main/java/org/jboss/as/controller/ListAttributeDefinition.java

```diff
@@ -184,6 +193,10 @@ protected abstract void addOperationParameterValueTypeDescription(final ModelNod
                                                                       final ResourceDescriptionResolver resolver,
                                                                       final Locale locale, final ResourceBundle bundle);
 
+    protected abstract void addOperationReplyValueTypeDescription(final ModelNode node, final String operationName,
+                                                                      final ResourceDescriptionResolver resolver,
+                                                                      final Locale locale, final ResourceBundle bundle);
+
```

Comment:
This and the new method in MapAttributeDefinition concern me a bit since they will break subclasses. This won't go in until core 3.0, and that's a new major, so technically that's ok. But there's such a high chance that this will end up in a minor version of some downstream software, e.g. a WildFly 10.1 or an EAP 7.1, and then we'll have hassles if we break external extensions (e.g. Teiid, JDV).

WDYT about having these by default call addOperationParameterValueTypeDescription ? Perhaps with a TODO pointing at a JIRA to get rid of that eventually (i.e. for EAP 8.) At a glance it looks like the impls you added here and in the full wildfly PR end up executing the same code as addOperationParameterValueTypeDescription.